### PR TITLE
feat: [#185555573] state contracting quick action

### PR DESCRIPTION
**Pull request by FarazA22**
### Ticket

https://www.pivotaltracker.com/story/show/186150128

### Approach
Separate quick action tasks that are used to populate the pages from quick actions that are for external links

## Code author checklist

- [ ] I have performed a self-review of my code
- [ ] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [ ] I have not used any relative imports
- [ ] I have checked for and removed instances of unused config from CMS
- [ ] I have pruned any instances of unused code
- [ ] I have not added any markdown to labels, titles and button text in config
- [ ] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [ ] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

File: web/src/lib/static/mockHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import fs from "fs";
+
+export const mockReadDirReturn = (value: string[], mockedFs: jest.Mocked<typeof fs>): void => {
```
Comment (somabadri) [inline web/src/lib/static/mockHelpers.ts:3]: [pebble/dust] can we parameterize these arguments so the function can be called something along the lines of 
```
mockReadDirReturnOnce({
  mockedFs, 
  values/returnValues: ["license1.md", "license2.md"]
});
```
to improve readability?

File: web/src/pages/dashboard.tsx [hunk 2]
```diff
@@ -162,15 +167,48 @@ const DashboardPage = (props: Props): ReactElement => {
...
+          key={registryUpdateBrcAmendment.filename}
+        />
+      );
+    }
+
+    const nonHicStateContracting = props.quickActionLinks.find(
+      (e) => e.filename === "state-contracting-external-link"
+    );
+
+    if (nonHicStateContracting && !isHomeContractorIndustry(business?.profileData.industryId)) {
```
Comment (somabadri) [inline web/src/pages/dashboard.tsx:190]: [pebble] thoughts on extracting the second half of the conditional into it's own variable so it's easier to read? 
`if(nonHicStateContracting && isNotHomeContractor)`

Comment (somabadri) [reply web/src/pages/dashboard.tsx:190]: or even extracting the whole conditional and naming it something along the lines of `renderNonHicStateContracting`so it's `if(renderNonHicStateContracting){...}`

File: web/src/lib/static/loadMunicipalities.test.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import { generateMunicipalityDetail } from "@businessnjgovnavigator/shared";
-import fs from "fs";
+import fs from "fs"; // Object
 import { loadAllMunicipalities } from "./loadMunicipalities";
 
-jest.mock("fs");
+jest.mock("fs"); // Object with Jest Fns
```
Comment (AnJuHyppolite) [inline web/src/lib/static/loadMunicipalities.test.ts:5]: We are importing `fs` and using `jest.mock("fs");` in multiple places throughout our codebase, why comment these now?

[pebble] remove commented code

Comment (FarazA22) [reply web/src/lib/static/loadMunicipalities.test.ts:5]: Good catch, these comments were used to further explain the concepts of mocking, I didn't intend them to be there

**Pull request merged**